pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3191,16 +3191,41 @@ func (c *Controller) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 		return vm, common.NewSyncError(fmt.Errorf(fetchingRunStrategyErrFmt, err), FailedCreateReason), err
 	}
 
-	// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
-	err = c.instancetypeMethods.StoreControllerRevisions(vm)
-	if err != nil {
-		log.Log.Object(vm).Infof("Failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
-		c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
-		return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+	// TODO(lyarwood): Extract all instance type logic into a handler
+	switch c.clusterConfig.GetInstancetypeReferencePolicy() {
```

Comment:
nit: Can you add `default:` case to this `switch` and return an error, if the value is unexpected?

Reply:
We shouldn't need this as we should have enum validation in the openapi spec for the two supported values.

I'll add a functional test to ensure this.